core/src/main/java/org/elasticsearch/index/mapper/ip/IpFieldMapper.java

```diff
@@ -217,11 +220,58 @@ public Query rangeQuery(Object lowerTerm, Object upperTerm, boolean includeLower
             return InetAddressPoint.newRangeQuery(name(), lower, upper);
         }
 
+        /** Add both values and return the maximum value in case of overflow. */
+        private static byte[] add(byte[] a, byte[] b) {
+            byte[] aCopy = new byte[a.length + 1];
+            System.arraycopy(a, 0, aCopy, 1, a.length);
+            byte[] bCopy = new byte[b.length + 1];
+            System.arraycopy(b, 0, bCopy, 1, b.length);
+            byte[] sum = new byte[aCopy.length];
+            NumericUtils.add(aCopy.length, 0, aCopy, bCopy, sum);
+            if (sum[0] == 0) {
+                return Arrays.copyOfRange(sum, 1, sum.length);
+            } else {
+                byte[] result = new byte[a.length];
+                Arrays.fill(result, (byte) 0xff);
+                return result;
+            }
+        }
+
+        /** Subtract both values and return the minimum value in case of underflow. */
+        private static byte[] subtract(byte[] a, byte[] b) {
+            if (StringHelper.compare(a.length, a, 0, b, 0) <= 0) {
+                return new byte[a.length];
+            } else {
+                byte[] result = new byte[a.length];
+                NumericUtils.subtract(a.length, 0, a, b, result);
+                return result;
+            }
+        }
+
         @Override
         public Query fuzzyQuery(Object value, Fuzziness fuzziness, int prefixLength, int maxExpansions, boolean transpositions) {
-            InetAddress base = parse(value);
-            int mask = fuzziness.asInt();
-            return XInetAddressPoint.newPrefixQuery(name(), base, mask);
+            // TODO: this is crazy, throw an exception instead
```

Comment:
Can we do this instead (throw an exception)? And just tell the user to use a cidr mask? Supporting "fuzzy" queries on ip addresses seems crazy, and I don't think we should continue this.